third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -795,6 +795,14 @@ struct AtomicRMWOpConversion
     Type valueElemTy =
         tensorTy ? getTypeConverter()->convertType(tensorTy.getElementType())
                  : valueTy;
+
+    const bool doLLVMFallback =
+        valueElemTy.isBF16() && getNVIDIAComputeCapability(moduleOp) < 90;
+    auto llvmAtomicBinOp =
+        doLLVMFallback ? matchAtomicOp(atomicRmwAttr) : std::nullopt;
+    auto llvmAtomicMemOrdering =
+        doLLVMFallback ? getMemoryOrdering(op.getSem()) : std::nullopt;
```

Comment:
Given these are only used in the `doLLVMFallback` case, it feels like it might be a little cleaner to move these into that branch?  Sure, that moves it inside a loop, but I would be surprised if that has any real perf impact on the compiler